wkcuber/api/MagDataset.py

```diff
@@ -73,7 +73,7 @@ def read(
         offset: Tuple[int, int, int] = (0, 0, 0),
         size: Tuple[int, int, int] = None,
     ) -> np.array:
-        return self.view.read(offset, size)
+        return self.get_view().read(offset, size)
```

Comment:
@rschwanhold I was a bit confused by the test failure, but I was able to fix this with this change. I don't know the impact, but I think we need the correct offset => size calculation for the test to work.

Reply:
Oops, this seems to break more than it fixes. Can you have a look what the desired change should be?

Reply:
I will take a look at it

Reply:
The previous behavior was correct, but the test case was wrong.